HopsanGUI/HcomHandler.cpp

```diff
@@ -6439,154 +6442,126 @@ void HcomHandler::evaluateExpression(QString expr, VariableType desiredType)
+            mAnsType = Undefined;
+            return;
         }
 
-        QString dataVecArg;
-        QString timeVecArg;
-        QString typeArg;
-        QString windowingFuncArg;
-        QString minTimeArg;
-        QString maxTimeArg;
-
-        //Figure out which arguments mean what
-        dataVecArg = args[0];
-        if(args.size()>1) {
-            if(args[1] == "energy" || args[1] == "power" || args[1] == "rms") {
-                typeArg = args[1];
-            }
-            else {
-                timeVecArg = args[1];
-            }
-        }
-        if(args.size()>2) {
-          if(!timeVecArg.isEmpty()) {
-            typeArg = args[2];
-          }
-          else {
-            windowingFuncArg = args[2];
-          }
-        }
-        if(args.size() > 3) {
-            if(!timeVecArg.isEmpty()) {
-                windowingFuncArg = args[3];
-            }
-            else {
-                minTimeArg = args[3];
-            }
+        //Parse power spectrum type
+        FrequencySpectrumEnumT type;
+        if(isHcomFunctionCall("psd", expr)) {
+            type = PowerSpectrum;
         }
-        if(args.size() > 4) {
-            if(!timeVecArg.isEmpty()) {
-                minTimeArg = args[4];
-            }
-            else {
-                maxTimeArg = args[4];
-            }
+        else if(isHcomFunctionCall("esd", expr)) {
+            type = EnergySpectrum;
         }
-        if(args.size() > 5) {
-          maxTimeArg = args[5];
+        else if(isHcomFunctionCall("rmsd", expr)) {
+            type = RMSSpectrum;
         }
 
-        //Fetch data vector
-        evaluateExpression(dataVecArg, DataVector);
-        SharedVectorVariableT pDataVar = mAnsVector;
-        if (mAnsType != DataVector)
-        {
-            HCOMERR(QString("Variable: %1 was not found!").arg(dataVecArg));
+        //Data vector
+        int i=0;
+        const QString varName = splitArgs[i].trimmed();
+        evaluateExpression(varName, DataVector);
+        if(mAnsType != DataVector) {
+            HCOMERR(QString("Variable: %1 was not found!").arg(varName));
             mAnsType = Undefined;
             return;
         }
+        SharedVectorVariableT pVar = mAnsVector;
+        ++i;
 
-        //Fetch time vector
+        //Time vector
         SharedVectorVariableT pTimeVar;
-        if(timeVecArg.isEmpty())
-        {
-            pTimeVar = pDataVar->getSharedTimeOrFrequencyVector();
-        }
-        else
-        {
-            evaluateExpression(timeVecArg, DataVector);
-            pTimeVar = mAnsVector;
-            if (mAnsType != DataVector)
-            {
-                HCOMERR(QString("Variable: %1 was not found!").arg(timeVecArg));
-                mAnsType = Undefined;
-                return;
+        if(splitArgs.size() > i) {
+            QString arg2 = splitArgs[1].trimmed();
```

Comment:
Index i or 1 ? in this case I think i==1 but looks odd

Reply:
Fixed. It works with "1" too, but it is dangerous if someone decides to insert a new argument in the future.